File: parser-service/src/parse/dto/parse-result.dto.ts
export class ParseResultDto {
  // 原始文本内容
  rawText: string;

  // 提取的结构化信息
  name: string;
  phone: string;
  email: string;
  workYears: number;
  skills: string[];
  education: string;

  // 解析置信度 (0-100)
  confidence: number;

  // 解析方法说明
  parseMethod: string;

  // 错误信息（如果有）
  error?: string;
}

File: backend/src/main/java/com/recruit/dto/ParseResultDTO.java
package com.recruit.dto;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;
import java.util.Map;

@Data
public class ParseResultDTO {
    private String name;
    private String phone;
    private String email;
    private Integer workYears;
    private List<String> skills;

    @JsonAlias({"confidenceScore", "confidence"})
    private BigDecimal confidenceScore;

    @JsonAlias({"rawData", "rawText"})
    private Object rawData;

    private String error;

    @JsonProperty("confidence")
    public BigDecimal getConfidence() {
        return confidenceScore;
    }

    @JsonProperty("confidence")
    public void setConfidence(BigDecimal confidence) {
        this.confidenceScore = confidence;
    }
}
